deebotozmo/cli.py

```diff
@@ -0,0 +1,542 @@
+) -> None:
+    """Click subcommand that returns the live map."""
+    if not force_extension and mimetypes.guess_type(filepath)[0] != "image/png":
+        logging.error("exportlivemap generates a png image.")
+        logging.error(
+            "either change your file extension to 'png' or pass '--force_extension'."
+        )
+        return
+
+    vacbot = CliUtil()
+    try:
+        await vacbot.before(ctx.obj["DEVICE"])
+
+        lock = asyncio.Event()
+
+        async def on_map(_: MapEvent) -> None:
+            with open(filepath, "wb") as file:
+                file.write(base64.decodebytes(vacbot.bot.map.get_base64_map()))
+            lock.set()
+
+        lock.clear()
+        listener = vacbot.bot.events.map.subscribe(on_map)
+        await lock.wait()
+        listener.unsubscribe()
+    finally:
+        await vacbot.after()
+
+
+@cli.command(name="getdevices", help="Get Devices")
+@click.option(
+    "--raw-data/--no-raw-data", default=False, help="Return raw data about all devices"
+)
+@coro
+async def get_devices(raw_data: bool) -> None:
+    """Click subcommand that returns all devices."""
+    vacbot = CliUtil()
+    try:
+        await vacbot.before()
+        if raw_data:
+            print(vacbot.devices)
+        else:
+            for idx, device in enumerate(vacbot.devices):
+                print(
+                    f"{idx + 1}) {device.nick} ({device.device_name}) ({device.did}) ({device.name})"
+                )
+    finally:
+        await vacbot.after()
+
+
+class CliUtil:
+    """CliUtil communicates with the bot to provide self.bot."""
+
+    def __init__(self) -> None:
+        """Init CliUtil."""
+        self._config = read_config()
+
+        self._session = aiohttp.ClientSession()
+
+        self._api = EcovacsAPI(
+            self._session,
+            self._config["device_id"],
+            self._config["email"],
+            self._config["password_hash"],
+            continent=self._config["continent"],
+            country=self._config["country"],
+            verify_ssl=bool(self._config["verify_ssl"]),
+        )
+
+        if not config_file_exists():
+            click.echo(
+                f"Not logged in. Do '{os.path.basename(sys.argv[0])} createconfig' first."
+            )
+            sys.exit(1)
+
+    async def before(self, match: Union[str, None] = None) -> None:
+        # pylint: disable=attribute-defined-outside-init
+        """Communicate with Deebot."""
+        await self._api.login()
+
+        self.devices = await self._api.get_devices()
+        if match is None:
+            device = self.devices[0]
+        else:
+            device_ = self.match_device(match)
+            if device_ is None:
+                logging.warning(
+                    "Failed to find a device, defaulting to first item in list."
+                )
+                device = self.devices[0]
+            else:
+                device = device_
```

Comment:
```suggestion
        device = self._get_matched_device(device)
```

makes your code simpler